WIP: Documentation

GDYendell opened the pull request:
Remove ini file configuration
Remove command line options that were added for testing before we had JSON configuration
Add some documentation on configuration of processes and plugins

GDYendell commented:
The build failure seems unrelated to these changes: https://travis-ci.org/odin-detector/odin-data/builds/581136954#L9943

timcnicholls commented:
> The build failure seems unrelated to these changes: https://travis-ci.org/odin-detector/odin-data/builds/581136954#L9943

Build failure is due to the problem documented in #216 due to Travis switching to Xenial. It's fixed with a commit to the integration-test branch.

timcnicholls commented:
Thanks for taking this on @GDYendell ! 

Couple of starting observations:

1. The opening paragraph still refers to FrameReceiver as a single item rather than odin-data as a framework with multiple apps. We should fix this too. 
2. You're welcome to steal some of the content from the workshop notes I wrote for my group on building and using odin-data [here](https://github.com/stfc-aeg/odin-workshop/blob/master/odin-data.md)
3. Lots of the python tools are obsolete and need removing
4. The frameSimulator is becoming a first-class app in odin-data and should be documented too.

As a final note, I'd like to make a gentle reminder that we agreed about a month ago not to use PRs for WIP on branches. It causes difficulty making sense of the chronology of PRs when tagging releases and can become stale when requiring repeated reviews. I'd suggest that, if input and collaboration is required on such activities, push the WIP to a branch on this repo (rather than e.g `dls-controls`), open an issue and tag others in the issue comments to request input. We can add a 'for review' label on issues to flag these.

GDYendell commented:
> The opening paragraph still refers to FrameReceiver as a single item rather than odin-data as a framework with multiple apps. We should fix this too.

Added a new opening paragraph.

> You're welcome to steal some of the content from the workshop notes I wrote for my group on building and using odin-data here

Those notes are really useful, I have stolen some bits.

> Lots of the python tools are obsolete and need removing

I have removed the docs for those that have already been removed. The only things left that I think might be obselete are frame_producer and frame_receiver/client. I will remove these too if no one has any complaints.

> The frameSimulator is becoming a first-class app in odin-data and should be documented too.

Added a bit about FrameSimulator but it needs more. Maybe you could add to this @skyfrench?

> push the WIP to a branch on this repo (rather than e.g dls-controls), open an issue and tag others in the issue comments to request input.

Shall I do this now and close the PR? I expect this will go on for a couple of weeks with, hopefully, various people's input.

timcnicholls commented:
> Shall I do this now and close the PR? I expect this will go on for a couple of weeks with, hopefully, various people's input.

Yes please. There's a "Help Wanted" label for issues already, which you can use to flag input needed on a WIP branch.
